activerecord/lib/active_record/scoping/named.rb

```diff
@@ -30,13 +30,8 @@ def all
         end
 
         def default_scoped # :nodoc:
-          scope = build_default_scope
-
-          if scope
-            relation.spawn.merge!(scope)
-          else
-            relation
-          end
+          scope = relation
+          build_default_scope(scope) || scope
```

Comment:
`ActiveRecord::Core#relation` creates new `Relation` instance each time.
https://github.com/rails/rails/blob/v5.1.1/activerecord/lib/active_record/core.rb#L307-L315

And also `build_default_scope` doesn't mutate `base_rel` because using `Relation#merge` (not `Relation#merge!`) for now.

So looks safe I think.